system-int-test/test/cmr/system_int_test/search/humanizer/community_usage_metrics_test.clj

```diff
@@ -221,103 +221,102 @@
-                                     column-name)]}
-                   (humanizer-util/update-community-usage-metrics admin-update-token csv)))
+        (is (= {:status 422
+                :errors [(format "A '%s' column is required in community usage CSV data"
+                                 column-name)]}
+               (humanizer-util/update-community-usage-metrics admin-update-token csv)))
 
-            "Missing product (short-name)"
-            "Product" "Version,Hosts\n3,4"
+        "Missing product (short-name)"
+        "Product" "ProductVersion,Hosts\n3,4"
 
-            "Missing hosts (access-count)"
-            "Hosts" "Product,Version\nAMSR-L1A,4"))
+        "Missing hosts (access-count)"
+        "Hosts" "Product,ProductVersion\nAMSR-L1A,4"))
 
     (testing "Empty Required CSV Column"
       (testing "Empty Product"
         (is (= {:status 422
                 :errors ["Error parsing 'Product' CSV Data. Product may not be empty."]}
                (humanizer-util/update-community-usage-metrics
                 admin-update-token
-                "Product,Version,Hosts\n,4,64"))))
+                "Product,ProductVersion,Hosts\n,4,64"))))
       (testing "Empty Hosts"
         (is (= {:status 422
-                :errors ["Error parsing 'Hosts' CSV Data for collection [AMSR-L1A], version [4]. Hosts may not be empty."]}
+                :errors ["Error parsing 'Hosts' CSV Data. Hosts may not be empty. Csv line entry: [\"AMSR-L1A\" \"4\" \"\"]"]}
                (humanizer-util/update-community-usage-metrics
                 admin-update-token
-                "Product,Version,Hosts\nAMSR-L1A,4,")))))
+                "Product,ProductVersion,Hosts\nAMSR-L1A,4,")))))
 
     (testing "Maximum product length validations"
       (let [long-value (apply str (repeat 86 "x"))]
         (is (= {:status 400
                 :errors ["#/0/short-name: expected maxLength: 85, actual: 86"]}
                (humanizer-util/update-community-usage-metrics
                 admin-update-token
-                (format "Product,Version,Hosts\n%s,3,4" long-value))))))
+                (format "Product,ProductVersion,Hosts\n%s,3,4" long-value))))))
 
-    (testing "Maximum version length validations"
-      (let [long-value (apply str (repeat 21 "x"))]
-        (is (= {:status 400
-                :errors ["#/0/version: expected maxLength: 20, actual: 21"]}
-               (humanizer-util/update-community-usage-metrics
-                admin-update-token
-                (format "Product,Version,Hosts\nAST_09XT,%s,4" long-value))))))
+    ;; (testing "Maximum version length validations"
+    ;;   (let [long-value (apply str (repeat 21 "x"))]
+    ;;     (is (= {:status 400
+    ;;             :errors ["#/0/version: expected maxLength: 20, actual: 21"]}
+    ;;            (humanizer-util/update-community-usage-metrics
+    ;;             admin-update-token
+    ;;             (format "Product,Version,Hosts\nAST_09XT,%s,4" long-value))))))
```

Comment:
If this test is OBE it should be removed, if it's still valid we should leave it in and enabled

Reply:
Do you think the solution I chose is reasonable? I wasn't sure if that was the best way to go about it. I'm not really sure why there is this 20 character limit at all.  I could also remove the limit from the schema, but not sure if that would affect anything (possibly Earthdata Search can't support versions longer than 20 characters?)

I made the decision that if an item has a Version longer than 20 characters, I would drop the version for that product. This causes products that have versions longer than 20 characters to be summed over the entire product instead of by version. The rankings will still work without the version.